core/src/main/java/com/linecorp/armeria/internal/server/annotation/AnnotatedValueResolver.java

```diff
@@ -818,6 +859,62 @@ private static AnnotatedValueResolver ofInjectableTypes0(String name, AnnotatedE
         };
     }
 
+    /**
+     * Returns an attribute resolver which retrieves a value specified by {@code attrKeys}
+     * from the {@link RequestContext}.
+     */
+    private static BiFunction<AnnotatedValueResolver, ResolverContext, Object>
+    attributeResolver(Iterable<AttributeKey<?>> attrKeys) {
+        return (resolver, ctx) -> {
+            final StringBuilder errorMsgBuilder = new StringBuilder();
+            for (AttributeKey<?> attrKey : attrKeys) {
+                final Object value = ctx.context.attr(attrKey);
+                if (value != null) {
+                    final boolean isValidType;
+
+                    final Class<?> rawContainerType = resolver.rawContainerType();
+                    if (rawContainerType != null) {
+                        isValidType = rawContainerType.isInstance(value);
+                        if (!isValidType) {
+                            errorMsgBuilder.append("The value ");
+                            errorMsgBuilder.append(value);
+                            errorMsgBuilder.append(" cannot be cast to rawContainerType ");
+                            errorMsgBuilder.append(rawContainerType);
+                        }
+                    } else {
+                        final Class<?> elementType = resolver.elementType().isPrimitive() ?
+                                                     Primitives.wrap(resolver.elementType())
+                                                     : resolver.elementType();
+
+                        isValidType = elementType.isInstance(value);
+                        if (!isValidType) {
+                            errorMsgBuilder.append("The value ");
+                            errorMsgBuilder.append(value);
+                            errorMsgBuilder.append(" cannot be cast to elementType ");
+                            errorMsgBuilder.append(elementType);
+                        }
+                    }
+
+                    if (isValidType) {
+                        return value;
+                    } else {
+                        errorMsgBuilder.append(". For your information, attrKey Name is ");
+                        errorMsgBuilder.append(attrKey.name());
+                        errorMsgBuilder.append(", httpElementName is ");
+                        errorMsgBuilder.append(resolver.httpElementName());
+                        errorMsgBuilder.append(". ");
+                    }
+                }
+            }
+
+            if (!StringUtil.isNullOrEmpty(errorMsgBuilder.toString())) {
+                throw new ClassCastException(errorMsgBuilder.toString());
+            } else {
+                return resolver.defaultOrException();
```

Comment:
We should probably also update the error message here

https://github.com/line/armeria/blob/3d800a2be414723d9c49abfd9db95c037e822559/core/src/main/java/com/linecorp/armeria/internal/server/annotation/AnnotatedValueResolver.java#L1055

Reply:
@jrhee17 nim, thanks for your comment. 
i missed your comments 😢 , because no alarm and no display on my github pages 🙇‍♂️ 

Yes, i modify it! 
- As-is : `parameter/header` 
- To-Be :`parameter/header/attribute`.